compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -411,6 +413,18 @@ impl SirFuncCx<'tcx> {
         ykpack::Local(local.as_u32())
     }
 
+    fn lower_cast_misc(&self, op: &mir::Operand<'_>, ty: Ty<'_>) -> ykpack::Rvalue {
+        let lop = self.lower_operand(op);
+        let lty = match ty.kind() {
+            ty::Int(si) => lower_signed_int(*si),
+            ty::Uint(ui) => lower_unsigned_int(*ui),
+            ty::Bool => ykpack::Ty::Bool,
+            ty::Char => ykpack::Ty::Char,
+            _ => ykpack::Ty::Unimplemented(format!("{:?}", ty)),
```

Comment:
This can also go from enum to discriminant value, from fat ptr to fat ptr and from fat ptr to thin ptr. It is fine to leave that unimplemented for now though.

Reply:
Yeah. If it crashes, then I'm good with that.

Reply:
Ah, good to know. I assumed there are other casts, but those were the ones I cared about most to get us going.